build/scripts/ensure-toolset.ps1

```diff
@@ -0,0 +1,58 @@
+param ()
+set-strictmode -version 2.0
+$ErrorActionPreference="Stop"
+
+$nugetVersion = "3.6.0-beta1"
+$msbuildToolsetVersion = "0.1.3-alpha"
+$repoDir = [IO.Path]::GetFullPath((join-path $PSScriptroot "..\.."))
+$toolsDir = join-path $repoDir "Binaries\Toolset"
+
+# Download the specified tool.  The download name needs to be a version name that
+# will be different when we change the version strings above.  That way the script
+# does work when the strings change vs. using the already downloaded and cached
+# values.
+function download-tool() {
+    param ([string]$url, [string]$fileName, [string]$downloadName)
+
+    $scratchDir = join-path $toolsDir "Scratch"
+    mkdir $scratchDir -ErrorAction SilentlyContinue | out-null
+    
+    $filePath = join-path $toolsDir $fileName
+    $downloadPath = join-path $scratchDir $downloadName
+
+    if ((test-path $filePath) -and (test-path $downloadPath)) {
+        return $false
+    }
+        
+    write-host "Downloading $fileName"
+    $webClient = New-Object -TypeName "System.Net.WebClient"
+    $webClient.DownloadFile($url, $downloadPath)
+    cp $downloadPath $filePath
+    return $true
+}
+
+try {
+    mkdir $toolsDir -ErrorAction silent | out-null
+    pushd $toolsDir
+    
+    $nugetUrl = "https://dist.nuget.org/win-x86-commandline/v$nugetVersion/NuGet.exe"
+    if (download-tool $nugetUrl "NuGet.exe" "NuGet-$nugetVersion.exe") {
+        cp "NuGet.exe" $repoDir
+    }
+
+    $msbuildUrl = "https://jdashstorage.blob.core.windows.net/msbuild/msbuild-$msbuildToolsetVersion.zip" 
```

Comment:
@tmat 

Debugger I'm less sure of.  I know they shell out to tools at least.  But that's not an issue.  

For SDK just look at the NuGet package we consume.  It's an MSBuild task that contains a lot of VS dependencies (Setup in particular).  Additionally other DLLS like MS.Vs.Settings which is C++ IIRC.  Chances of that being made portable are very low. 

Don't forget about WPF. 

Also need to account for future VS dependencies in MSBuild.  The VS team has very little incentive to produce portable tasks.  Every future piece of tooling will be a fight to get them onto CLI.